src/base/types/Reaction.jl

```diff
@@ -50,24 +51,95 @@ mutable struct Reaction
         annotations,
         objective_coefficient,
     )
+end
 
-    # this constructor is only used internally
-    function Reaction(
-        id::String,
-        metabolites::Dict{String,Float64},
-        dir = :bidirectional;
-        default_bound = _constants.default_reaction_bound,
-    )
-        if dir == :forward
-            lb = 0.0
-            ub = default_bound
-        elseif dir == :reverse
-            lb = -default_bound
-            ub = 0.0
-        else
-            lb = -default_bound
-            ub = default_bound
-        end
-        Reaction(id; metabolites = metabolites, lb = lb, ub = ub)
+"""
+Reaction(
+    id::String,
+    metabolites::Dict{String,Union{Int, Float64}},
+    dir = :bidirectional;
+    default_bound = _constants.default_reaction_bound,
+)
+
+Convenience constructor for `Reaction`.
```

Comment:
maybe add the fact that this one is created from a dictionary of metabolites and the second one from a vector